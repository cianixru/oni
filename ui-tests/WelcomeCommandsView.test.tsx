import { shallow } from "enzyme"
import toJson from "enzyme-to-json"

import * as React from "react"

import {
    WelcomeButton,
    WelcomeCommandsView,
} from "./../browser/src/Editor/NeovimEditor/WelcomeBufferLayer"

describe("Welcome Layer test", () => {
    const buttons = [
        "button1",
        "button2",
        "button3",
        "button4",
        "button5",
        "button6",
        "button7",
        "button8",
    ]

    const commands = {
        openFile: { command: "button1" },
        openTutor: { command: "button2" },
        openDocs: { command: "button3" },
        openConfig: { command: "button4" },
        openThemes: { command: "button5" },
        openWorkspaceFolder: { command: "button6" },
        commandPalette: { command: "button7" },
        commandline: { command: "button8" },
    }
    const executeMock = jest.fn()
    it("should render without crashing", () => {
        const wrapper = shallow(
            <WelcomeCommandsView
                selectedId="button1"
                active
                commands={commands}
                executeCommand={executeMock}
            />,
        )
        expect(wrapper.length).toBe(1)
    })

    it("should match last snapshot on record", () => {
        const wrapper = shallow(
            <WelcomeCommandsView
                selectedId="button1"
                active
                commands={commands}
                executeCommand={executeMock}
            />,
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    it("should have the correct number of buttons for each commands", () => {
        const wrapper = shallow(
            <WelcomeCommandsView
                selectedId="button1"
                active
                commands={commands}
                executeCommand={executeMock}
            />,
        )
        expect(wrapper.dive().find(WelcomeButton).length).toBe(Object.values(commands).length)
    })
})
