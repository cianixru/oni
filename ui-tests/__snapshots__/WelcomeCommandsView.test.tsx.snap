// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Layer test should match last snapshot on record 1`] = `
<styled.div>
  <styled.div
    duration="0.25s"
  >
    <styled.div>
      Quick Commands
    </styled.div>
    <WelcomeButton
      command="button1"
      description="Control + N"
      onClick={[Function]}
      selected={true}
      title="New File"
    />
    <WelcomeButton
      command="button6"
      description="Control + O"
      onClick={[Function]}
      selected={false}
      title="Open File / Folder"
    />
    <WelcomeButton
      command="button7"
      description="Control + Shift + P"
      onClick={[Function]}
      selected={false}
      title="Command Palette"
    />
    <WelcomeButton
      command="editor.openExCommands"
      description=":"
      onClick={[Function]}
      selected={false}
      title="Vim Ex Commands"
    />
  </styled.div>
  <styled.div
    duration="0.25s"
  >
    <styled.div>
      Learn
    </styled.div>
    <WelcomeButton
      command="button2"
      description="Learn modal editing with an interactive tutorial."
      onClick={[Function]}
      selected={false}
      title="Tutor"
    />
    <WelcomeButton
      command="button3"
      description="Discover what Oni can do for you."
      onClick={[Function]}
      selected={false}
      title="Documentation"
    />
  </styled.div>
  <styled.div
    duration="0.25s"
  >
    <styled.div>
      Customize
    </styled.div>
    <WelcomeButton
      command="button4"
      description="Make Oni work the way you want."
      onClick={[Function]}
      selected={false}
      title="Configure"
    />
    <WelcomeButton
      command="button5"
      description="Choose a theme that works for you."
      onClick={[Function]}
      selected={false}
      title="Themes"
    />
  </styled.div>
</styled.div>
`;
